modules/internal/ChapelLocale.chpl

```diff
@@ -448,11 +577,22 @@ module ChapelLocale {
     iter chpl_initOnLocales() {
       if numLocales > 1 then
         halt("The locales must be initialized in parallel");
-      for locIdx in (origRootLocale:borrowed RootLocale?)!.getDefaultLocaleSpace() {
+      for locIdx in (origRootLocale._instance:borrowed RootLocale?)!.getDefaultLocaleSpace() {
         yield locIdx;
         rootLocale = origRootLocale;
         rootLocaleInitialized = true;
       }
+
+      // currently, standalone version of this iterator is not called, but just
+      // to be sure, at the end, visit all locales and set their locale-private
+      // `doneCreatingLocales`
+      for locIdx in 0..#numLocales /*ref(b)*/ {
+        on __primitive("chpl_on_locale_num",
+                       chpl_buildLocaleID(locIdx:chpl_nodeID_t,
+                                          c_sublocid_any)) {
+          doneCreatingLocales = true;
```

Comment:
We halt if `numLocales > 1` above, so I don't think this is needed. 

There is also an existing locale-private var `rootLocaleInitialized`, which I think should fulfill the same function as `doneCreatingLocales`. On master I don't see `doneCreatingLocales` ever being set, so I wonder if we can just remove it and rely on `rootLocaleInitialized` for the cases you need it.